include/hipSYCL/sycl/libkernel/accessor.hpp

```diff
@@ -111,6 +111,18 @@ constexpr accessor_variant deduce_accessor_variant(mode_tag<TagDescT> tag,
   }
 }
 
+#if defined(_WIN32) && defined(__clang__) && defined(__CUDA__)
```

Comment:
Might also add `__HIP__` there. There's no ROCm on Windows yet, but in principle it should be affected by the same issue.